unity/PingPongGameExample/PingPongGame/Assets/Scripts/NetTest.cs

```diff
@@ -15,42 +15,100 @@
  * limitations under the License.
  */
 
-using System.Collections;
 using System.Collections.Concurrent;
 
 using System.Net.Sockets;
+using System.Net.Http;
+using System.Net.NetworkInformation;
 using System.Diagnostics;
 using System.Threading;
-using System.Threading.Tasks;
 
 using System;
 
 public class NetTest
 {
-  public class HostAndPort
+  public enum TestType
+  {
+    PING = 0,
+    CONNECT = 1,
+  };
+
+  public class Site
   {
     public string host;
     public int port;
-    public double lastPingMs = 0f;
+    public string L7Path; // This may be load balanced.
+    public double lastPingMs;
+
+    public TestType testType;
+
+    int idx;
+    int size;
```

Comment:
Is this the size of samples? It is never initialized in the constructor

Reply:
It's initialized to zero implicitly (and a compiler message otherwise). Add sample goes up from there until array size.